src/snmalloc/mem/localalloc.h

```diff
@@ -767,6 +768,9 @@ namespace snmalloc
 #ifdef SNMALLOC_PASS_THROUGH
       return external_alloc::malloc_usable_size(const_cast<void*>(p_raw));
 #else
+
+      if (SecondaryAllocator::has_secondary_ownership(p_raw))
```

Comment:
I would prefer this to check that the allocation is not owned by snmalloc, rather than rely on the secondary allocator having this.  For GWP Asan, it is fine.

But I would also been keen to use this to build an approach on Windows to replace the allocator, and pass any original frees back to the Windows Heap.

Reply:
```
    SNMALLOC_FAST_PATH size_t alloc_size(const void* p_raw)
    {
#ifdef SNMALLOC_PASS_THROUGH
      return external_alloc::malloc_usable_size(const_cast<void*>(p_raw));
#else

      if (!check_domestication(p_raw))
        return SecondaryAllocator::alloc_size(p_raw);

  }
```

Could you illustrate a better way to do this? If do it the way above, we will always have the overhead for domestication test.

Reply:
```C++
      const PagemapEntry& entry =
        Config::Backend::get_metaentry(address_cast(p_raw));

      // Check is managed by snmalloc
      if (SNMALLOC_LIKELY(entry.get_remote() != nullptr)
        return sizeclass_full_to_size(entry.get_sizeclass());

      // If not managed by snmalloc, pass to secondary allocator.
      return SecondaryAllocator::alloc_size(p_raw);
```